dll/win32/shell32/CDefView.cpp

```diff
@@ -607,11 +618,31 @@ void CDefView::UpdateStatusbar()
             pIcon = (LPARAM)m_hMyComputerIcon;
         }
 
+        if (hThread != m_hUpdateStatusbarThread)
+            return;
+
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETICON, 2, pIcon, &lResult);
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 2, (LPARAM)szPartText, &lResult);
     }
 }
 
+static unsigned __stdcall UpdateStatusbarProc(void *args)
```

Comment:
This should be a static method of CDefView. This would certainly allow you to access the `m_hUpdateStatusbarThread` member of `pView` even if that member was private (or protected?)

Reply:
OK. Thanks.

Reply:
Renamed as `_UpdateStatusbarProc`.